### PR TITLE
Re-write SS018 using IOperation

### PR DESCRIPTION
**Pull request by Youssef1313**
This might improve the performance (#148).

@Vannevelj Could you confirm if this fixes the performance problem?

Either way, using `IOperation` is simpler and better here.

### CODE REVIEW

File: SharpSource/SharpSource/Diagnostics/SwitchDoesNotHandleAllEnumOptionsAnalyzer.cs [hunk 1]
```diff
@@ -27,35 +25,29 @@ public override void Initialize(AnalysisContext context)
...
+        foreach (var caseOperation in switchOperation.Cases)
         {
-            foreach (var label in section.Labels)
+            foreach (var caseClauseOperation in caseOperation.Clauses)
             {
-                if (label.IsKind(SyntaxKind.CaseSwitchLabel))
+                if (caseClauseOperation is ISingleValueCaseClauseOperation { Value: IFieldReferenceOperation enumReference })
                 {
-                    var switchLabel = (CaseSwitchLabelSyntax)label;
-                    var symbol = context.SemanticModel.GetSymbolInfo(switchLabel.Value).Symbol;
```
Comment (Youssef1313) [inline SharpSource/SharpSource/Diagnostics/SwitchDoesNotHandleAllEnumOptionsAnalyzer.cs:50]: I'm suspecting `GetSymbolInfo` call here is expensive and is the reason of the performance issue.

**Comment by Vannevelj** [comment]
Another interesting change! I'll merge this regardless but I'll wait with it a day or two until I get home and can verify whether this addressed the performance issue. Thanks 🙏

**Comment by Youssef1313** [comment]
> Another interesting change! I'll merge this regardless but I'll wait with it a day or two until I get home and can verify whether this addressed the performance issue. Thanks 🙏

Yeah it would be good to verify before merging. Note: `GetSymbolInfo` was previously shown as being slow in https://github.com/dotnet/roslyn-analyzers/issues/3221

**Comment by Vannevelj** [comment]
I haven't noticed an improvement when playing around with the previous problematic case. However that is in itself in line with https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0010 -- both analyzers do the same thing and can never seem to generate the code fix preview. Presunably the performance is related to the CodeFix implementation and not the analyzer?

I'll leave the bug defect open for future investigation when I get to it but I'll merge this PR to at least get the `IOperation` already.

**Pull request merged**